oracle_cardano/chain/cardano_chain_info.go

```diff
@@ -0,0 +1,40 @@
+package chain
+
+import (
+	"context"
+
+	cCore "github.com/Ethernal-Tech/apex-bridge/oracle_common/core"
+	infracommon "github.com/Ethernal-Tech/cardano-infrastructure/common"
+)
+
+type CardanoChainInfo struct {
+	Config *cCore.CardanoChainConfig
+
+	ProtocolParams []byte
+}
+
+func NewCardanoChainInfo(config *cCore.CardanoChainConfig) *CardanoChainInfo {
+	return &CardanoChainInfo{
+		Config: config,
+	}
+}
+
+func (info *CardanoChainInfo) Populate(ctx context.Context) error {
```

Comment:
Maybe we should implement this with lazy loading
`protocolParams []byte`

func (info *CardanoChainInfo) GetProtocolParameters(ctx context.Context) ([]byte, error) {
   info.lock.Lock()
   defer info.lock.Unlock()
   if info.protocolParams != nil {
      return info.protocolParams
   }
   
   ...
}